### PR TITLE
LLM: using separate threads to do inference

### PR DESCRIPTION
**Pull request by WeiguangHan**
@hkvision Using separate threads to do inference.

### CODE REVIEW

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 1]
```diff
@@ -368,42 +383,28 @@ def run_transformer_int4_gpu(repo_id,
...
+            input_str = open(f"prompt/{test_length}.txt", 'r').read()
+            # As different tokenizer has different encodings,
+            # slice the input_ids to ensure the prompt length is required length.
+            input_ids = tokenizer.encode(input_str, return_tensors="pt")
+            input_ids = input_ids[:, :in_len]
+            true_str = tokenizer.batch_decode(input_ids)[0]
+            input_ids = tokenizer.encode(true_str, return_tensors="pt").to('xpu')
+            actual_in_len = input_ids.shape[1]
+            result[in_out] = []
+            for i in range(num_trials + warm_up):
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:404]: need to put the for loop in the thread as well.

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 2]
```diff
@@ -368,42 +383,28 @@ def run_transformer_int4_gpu(repo_id,
...
-                pass
+            in_out_len = in_out.split("-")
+            in_len = int(in_out_len[0])
+            out_len = int(in_out_len[1])
+            # As different tokenizer has different encodings,
+            # in_len.txt maybe shorter than we need,
+            # use much longer context to make sure input length
+            test_length = min(in_len*2, 8192)
+            while test_length not in [32, 256, 1024, 2048, 8192]:
+                test_length = test_length * 2
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:394]: need to check this logic, maybe in another PR.

File: python/llm/dev/benchmark/all-in-one/run.py (hunk 1 above)
Comment (WeiguangHan) [reply python/llm/dev/benchmark/all-in-one/run.py]: sure

File: python/llm/dev/benchmark/all-in-one/run.py (hunk 2 above)
Comment (WeiguangHan) [reply python/llm/dev/benchmark/all-in-one/run.py:395]: Will do it later.

**Review by hkvision — APPROVED** [review]

**Comment by hkvision** [comment]
At this moment, we first put model generate for target number of trials in a thread.
Later if want to further refine this, we may put the entire run_model in a thread.

**Pull request merged**